arangod/Replication/Syncer.cpp

```diff
@@ -559,23 +559,22 @@ Result Syncer::createCollection(TRI_vocbase_t* vocbase,
     // collection already exists. TODO: compare attributes
     return Result();
   }
-   
+
   bool forceRemoveCid = false;
   if (col != nullptr && simulate32Client()) {
     forceRemoveCid = true;
     LOG_TOPIC(TRACE, Logger::REPLICATION) << "would have got a wrong collection!";
     // go on now and truncate or drop/re-create the collection
   }
-  
+
   // conflicting collections need to be dropped from 3.3 onwards
-  col = vocbase->lookupCollection(name);
+  col = vocbase->lookupCollection(name).get();
+
```

Comment:
Why are you not converting this stuff to std::shared_ptr as well ?

Reply:
I tried to keep the changes to a minimum otherwise there are too many files in the changeset
although I do agree that eventually raw pointers should be shared pointers